python/lsst/cp/pipe/utils.py

```diff
@@ -58,3 +73,217 @@ def getTargetList(parsedCmd, **kwargs):
+        if raiseWithMessage:
+            msg = "Exposure lengths for visit pairs must be equal. " + \
+                  "Found %s and %s" % (expTime1, expTime2)
+            if v1 and v2:
+                msg += " for visit pair %s, %s" % (v1, v2)
+            raise RuntimeError(msg)
+        else:
+            return False
+    return True
+
+
+def validateIsrConfig(isrTask, mandatory=None, forbidden=None, desirable=None, undesirable=None,
+                      checkTrim=True, logName=None):
+    """Check that appropriate ISR settings have been selected for the task.
+
+    Note that this checks that the task itself is configured correctly rather
+    than checking a config.
+
+    Parameters
+    ----------
+    isrTask : `lsst.ip.isr.IsrTask`
+        The task whose config is to be validated
+
+    mandatory : `iterable` of `str`
+        isr steps that must be set to True. Raises if False or missing
+
+    forbidden : `iterable` of `str`
+        isr steps that must be set to False. Raises if True, warns if missing
+
+    desirable : `iterable` of `str`
+        isr steps that should probably be set to True. Warns is False, info if
+    missing
+
+    undesirable : `iterable` of `str`
+        isr steps that should probably be set to False. Warns is True, info if
+    missing
+
+    checkTrim : `bool`
+        Check to ensure the isrTask's assembly subtask is trimming the images.
+    This is a separate config as it is very ugly to do this within the
+    normal configuration lists as it is an option of a sub task.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if ``mandatory`` config parameters are False,
+        or if ``forbidden`` parameters are True.
+
+    TypeError
+        Raised if parameter ``isrTask`` is an invalid type.
+
+    Notes
+    -----
+    Logs warnings using an isrValidation logger for desirable/undesirable
+    options that are of the wrong polarity or if keys are missing.
+    """
+    if not isinstance(isrTask, ipIsr.IsrTask):
+        raise TypeError(f'Must supply an instance of ipIsr.IsrTask not {type(isrTask)}')
+
+    configDict = isrTask.config.toDict()
+
+    if logName and isinstance(logName, str):
+        log = lsstLog.getLogger(logName)
+    else:
+        log = lsstLog.getLogger("isrValidation")
+
+    if mandatory:
+        for configParam in mandatory:
+            if configParam not in configDict:
+                raise RuntimeError(f"Mandatory parameter {configParam} not found in the isr configuration.")
+            if configDict[configParam] is False:
+                raise RuntimeError(f"Must set config.isr.{configParam} to True for this task.")
+
+    if forbidden:
+        for configParam in forbidden:
+            if configParam not in configDict:
+                log.warn(f"Failed to find key {configParam} in the isr config. This is set to be " +
+                         "forbidden, so you should ensure that the equivalent for your obs_package to False.")
```

Comment:
If it's there, and you didn't set it to False, then fail.
If it's not there, then it can't be true, so that's kind of OK, but something might be fishy, so warn the user.
It's weird, but I think it's clear - I thought about this part reasonably carefully.

Reply:
There are reasons why some are `WARN`, some are `INFO` and some are `raise`. I think it's as it should be.

Reply:
I think the "it's in this list, but not in the Config" messages should make it more clear that that is where the discrepancy is (and there's also not an equivalent warning for the `required` list. Something like:

"Failed to find forbidden key {configParam} in the isr config. The keys in the forbidden list should each have an associated Field in IsrConfig: check that there is not a typo in this case."

Reply:
Done.